src/main/resources/blockly/js/mcreator_blockly.js

```diff
@@ -107,6 +107,43 @@ function appendDropDownWithMessage(messageKey, listType, fieldName) {
     };
 }
 
+// Helper function to use in Blockly extensions that register one dropdown to update contents of another
+// The block may define inputs called "<sourceName>Field" and "<targetName>Field" to customize fields' positions
+// Note that the source dropdown must be inserted before the target dropdown for their values to be loaded properly
+function appendAutoReloadingDropDown(sourceName, sourceList, targetName, targetList) {

```

Comment:
For those cases, field_data_list_selector.js should now be used

Reply:
I still don't see why you need any changes to this, and don't want to have changes here. It is also quite unrelated to the goal of the PR

Reply:
Hmm, this could be refactored to be specific to entity data parameters you mean?

Reply:
Well you can leave it as is and I can wait for community opinion, the opinion of SomeoneElse when/if he has time

Reply:
Checking again I think approach is mostly OK, we may have to do some fine-tuning when I get time to properly review this PR